### PR TITLE
In preparation for EXR, moved all texture loaders to deferred loading

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Engines/abstractEngine.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ import type { LoadFileError } from "../Misc/fileTools";
 import type { ShaderProcessingContext } from "./Processors/shaderProcessingOptions";
 import type { IPipelineContext } from "./IPipelineContext";
 import type { ThinTexture } from "../Materials/Textures/thinTexture";
-import type { IInternalTextureLoader } from "../Materials/Textures/internalTextureLoader";
+import type { IInternalTextureLoader } from "../Materials/Textures/Loaders/internalTextureLoader";
```
Comment (RaananW) [inline packages/dev/core/src/Engines/abstractEngine.ts]:
```suggestion
```

Fix the build issue. Seems to be not used.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by RaananW — COMMENTED** [review]
LGTM! Just a small question :-)

File: packages/dev/core/src/Materials/Textures/Loaders/internalTextureLoader.ts [hunk 2]
```diff
@@ -10,14 +10,6 @@ export interface IInternalTextureLoader {
      */
     supportCascades: boolean;
 
-    /**
-     * This returns if the loader support the current file information.
-     * @param extension defines the file extension of the file being loaded
-     * @param mimeType defines the optional mime type of the file being loaded
-     * @returns true if the loader can load the specified file
-     */
-    canLoad(extension: string, mimeType?: string): boolean;
```
[the remark this answers is not in the document]
Comment (deltakosh) [reply packages/dev/core/src/Materials/Textures/Loaders/internalTextureLoader.ts:19]: well it is all internal so no breaking change here ;)

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15458/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15458/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15458/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15458/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15458/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15458/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15458/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15458/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15458/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15458/merge/testResults/webgl2playwright/index.html

**Pull request merged**